app.rb

```diff
@@ -3,19 +3,60 @@
 require_relative './classes/student'
 require_relative './classes/teacher'
 require_relative './classes/rental'
+require 'json'
 
 class App
-  def initialize
-    @books = []
-    @people = []
-    @rentals = []
+  def initialize(books, people, rentals)
+    @books = books
+    @people = people
+    @rentals = rentals
+    # load people from json file during initialization
+    load_people
+    list_rentals
+  end
+
+  # Load books from JSON file
+  def load_books
+    if File.exist?('books.json')
+      books = JSON.parse(File.read('books.json'))
+      @books = books.map { |book| Book.new(book['title'], book['author']) }
+    else
+      @books = []
+    end
+  end
+
+  # Load people from JSON file
+  def load_people
+    if File.exist?('people.json')
+      persons = JSON.parse(File.read('people.json'))
+      @people = persons.map { |person| Person.new(person['age'], person['name']) }
+    else
+      @people = []
+    end
+  end
+
+  def list_rentals
+    if File.exist?('rentals.json')
+      rentals = JSON.parse(File.read('rentals.json'))
+      @rentals = rentals.map do |rental|
+        book = Book.new(rental['book']['title'], rental['book']['author'])
+        person = Person.new(rental['person']['age'], rental['person']['name'])
+        Rental.new(rental['date'], book, person)
+      end
+    else
+      @rentals = []
+    end
   end
```

Comment:
- [ ] According to the [project requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_preserve_data.md), on startup of the app we should load all data from files. .👀

- [ ] When I restarted your app, new `id`'s are generated for each persons which are different from the `id`'s save in the `JSON`  file

Example


Saved data
![image](https://github.com/shayan1234554321/school-library/assets/78251927/2f77aadc-0b5f-4e25-839a-5eb5ba9cc053)

Retrieved data
![image](https://github.com/shayan1234554321/school-library/assets/78251927/943a386e-faaa-4541-8d38-cc1099708308)

- [ ] Due to the difference in the `ids` it is impossible to `list_rentals`
![image](https://github.com/shayan1234554321/school-library/assets/78251927/f1466e0d-0a12-4253-9ff9-94f7873e111b)

![image](https://github.com/shayan1234554321/school-library/assets/78251927/3ac0df6b-504b-4703-9d19-a68d23da6d89)

- [ ] It would be nice if you include the `id`s among the retrieved data in the method `load_people` to avoid this error. 💪 🛠

Example
```
  def list_all_persons
    puts 'Database is empty! Add a person.' if persons.empty?
    persons.each { |person| puts "[#{person.class.name}] Name: #{person.name}, Age: #{person.age}, id: #{person.id}" }
  end
```